### ISSUE TITLE
Subscribe to queues without knowing the exchange and binding key (e.g. default exchange)

### ISSUE THREAD

**Issue by j-schreiber** [golevelup/nestjs#316]
Hey all, I am having real troubles understanding how golevelup expects me to know the exchange and routing key in advance, when I simply want to consume messages from a particular queue. As I understood it, the  `@RabbitSubscribe` decorator requires an `exchange` and `routingKey`, as shown in the example above:

```javascript
@RabbitSubscribe({
    exchange: 'exchange1',
    routingKey: 'subscribe-route',
    queue: 'subscribe-queue',
  })
  public async pubSubHandler(msg: {}) {
    console.log(`Received message: ${JSON.stringify(msg)}`);
  }
```

I understand, that golevelup dynamically creates the binding. But what happens, if there is no exchange? If any publisher publishes to `amq.direct` (which is also the "empty string"), I have a valid queue, but neither an exchange nor binding (because rabbit routes the message implicitly and the queue name actually is the binding).

How am I supposed to subscribe to a typical pub-sub model where multiple exchanges and bindings might publish to the same queue? I have a landscape where the individual consumers are not supposed to know about other publishers and also do not care about the RabbitMQ configuration that routes messages to "their" queues. As long as they consume from the queues, they do not need to know about other exchanges and bindings.

Any help appreciated!

**Comment by WonderPanda** [comment]
I think you probably want to use a `direct` exchange. https://www.rabbitmq.com/tutorials/tutorial-four-javascript.html

> The routing algorithm behind a direct exchange is simple - a message goes to the queues whose binding key exactly matches the routing key of the message.

I'm a bit confused about some of your other requirements though. It sounds like you're still required to know the names of queues ahead of time so there's still some burden of "knowing" the the destinations for things.

**Comment by j-schreiber** [comment]
Hey @WonderPanda  thanks for responding so quickly! Yes, the direct exchange is useful, but in most of my use cases it's not necessary. Instead, I would like to use the [default exchange](https://www.rabbitmq.com/tutorials/amqp-concepts.html#exchange-default) (which is either "empty string" or `amq.direct`). In that scenario, the publisher implicitly knows about the destination queue (even though it technically only specifies a binding key).

The queues are known ahead of time, yes. But to be precise, my question is: Can I subscribe to a queue without declaring an exchange and binding key? I simply want to consume everything from a particular queue, without worrying about the exchanges and bindings that route messages to that queue. Also, I do not want to filter. So I imagine something like this:

```javascript
@RabbitSubscribe({
    queue: 'my_queue_name',
})
public async pubSubHandler(msg: {}) {
    console.log(`Received message: ${JSON.stringify(msg)}`);
}
```

**Comment by WonderPanda** [comment]
Its not currently supported but I don't think it would be a big change. Are you interested in contributing this functionality? You would just need to update the decorator to make `exchange` and `routingKey` optional properties and then update the `setupSubscriberChannel` method in connection.ts to provide default values if no values are explicitly provided

**Comment by j-schreiber** [comment]
Absolutely. I am still a novice with NodeJS/Typescript, but I will try my best to prepare a PR.

**Comment by WonderPanda** [comment]
@j-schreiber No problem :) Feel free to ask if you have any questions and I can provide assistance

**Comment by weedget** [comment]
@WonderPanda do you have any plan/timeframe when you would like to prepare it? I'm also interested in this and maybe I can help somehow (for sure in testing this case).

**Comment by j-schreiber** [comment]
I plan to prepare the PR this week. We will see, if WonderPanda will approve it 😆 .

**Comment by WonderPanda** [comment]
Thanks for the contribution. These changes are published and available in `@golevelup/nestjs-rabbitmq@1.18.0`
